editor/plugins/texture_editor_plugin.cpp

```diff
@@ -82,25 +113,51 @@ void TexturePreview::_update_texture_display_ratio() {
 	}
 }
 
-void TexturePreview::_update_metadata_label_text() {
-	const Ref<Texture2D> texture = texture_display->get_texture();
+static Image::Format get_texture_2d_format(const Ref<Texture2D> &p_texture) {
+	// Why is format not just available in the Texture class?
 
-	String format;
-	if (Object::cast_to<ImageTexture>(*texture)) {
-		format = Image::get_format_name(Object::cast_to<ImageTexture>(*texture)->get_format());
-	} else if (Object::cast_to<CompressedTexture2D>(*texture)) {
-		format = Image::get_format_name(Object::cast_to<CompressedTexture2D>(*texture)->get_format());
-	} else {
-		format = texture->get_class();
+	const Ref<ImageTexture> image_texture = p_texture;
+	if (image_texture.is_valid()) {
+		return image_texture->get_format();
 	}
 
-	const Ref<Image> image = texture->get_image();
+	const Ref<CompressedTexture2D> compressed_texture = p_texture;
+	if (compressed_texture.is_valid()) {
+		return compressed_texture->get_format();
+	}
+
+	// AtlasTexture?
```

Comment:
Should be implemented.
CC @KoBeWi

Reply:
I didn't actually implement it here because the original code treated AtlasTexture differently too. But it depends what we would prefer, maybe it could use the format of the atlas.

Reply:
Using atlas' format makes sense.